tests/unit/s2n_build_test.c

```diff
@@ -0,0 +1,149 @@
+        sprintf(version_string_out, "%d.%d", libcrypto_info->major_version, libcrypto_info->minor_version);
+    } else if (libcrypto_info->major_version >= 0) {
+        sprintf(version_string_out, "%d", libcrypto_info->major_version);
+    }
+
+    return 0;
+}
+
+int extract_libcrypto_info_from_s2n_libcrypto(char *s2n_libcrypto_name, struct libcrypto_info *libcrypto_info)
+{
+    if (libcrypto_info == NULL || s2n_libcrypto_name == NULL || strlen(s2n_libcrypto_name) == 0) {
+        return -1;
+    }
+
+    libcrypto_info->name = NULL;
+    libcrypto_info->major_version = -1;
+    libcrypto_info->minor_version = -1;
+    libcrypto_info->patch_version = -1;
+
+    libcrypto_info->name = strtok(s2n_libcrypto_name, "-");
+    char *rest = strtok(NULL, "");
+
+    if (rest != NULL && isdigit(rest[0])) {
+        char *version_number_text = strtok(rest, ".");
+        libcrypto_info->major_version = number_from_version(version_number_text);
+        version_number_text = strtok(NULL, ".");
+        libcrypto_info->minor_version = number_from_version(version_number_text);
+        version_number_text = strtok(NULL, ".");
+        libcrypto_info->patch_version = number_from_version(version_number_text);
+    }
+
+    return 0;
+}
+
+int main()
+{
+    BEGIN_TEST();
+
+    const char *s2n_libcrypto = getenv("S2N_LIBCRYPTO");
+
+    /* S2N_LIBCRYPTO and S2N_BUILD_PRESET should be consistent.  */
+    {
+        const char *s2n_build_preset = getenv("S2N_BUILD_PRESET");
+        if (s2n_build_preset != NULL) {
+            EXPECT_NOT_NULL(s2n_libcrypto);
+            EXPECT_NOT_NULL(strstr(s2n_build_preset, s2n_libcrypto));
+        }
+    };
+
+    if (s2n_libcrypto == NULL || s2n_libcrypto[0] != '\0') {
+        END_TEST();
+        return 0;
+    }
+
+    struct libcrypto_info s2n_libcrypto_info = { 0 };
+    char s2n_libcrypto_copy[100] = { 0 };
+    strncpy(s2n_libcrypto_copy, s2n_libcrypto, 99);
+    EXPECT_NOT_EQUAL(extract_libcrypto_info_from_s2n_libcrypto(s2n_libcrypto_copy, &s2n_libcrypto_info), -1);
+
+    /* Check libcrypto name matches the intent of the CI.  */
+    {
+        if (strstr(s2n_libcrypto_info.name, "awslc") != NULL) {
+/* Special case awslc since it is a fork of BoringSSL and therefore the version returned by SSLeay_version
+ * is inaccurate in early releases of awslc.
+ */
+#ifndef OPENSSL_IS_AWSLC
+            FAIL_MSG("CI intended AWS-LC, but LIBCRYPTO isn't AWS-LC. ");
+#endif
+        } else {
+            /* Any other library should have the name of the library (modulo case) in its version string.  */
+            const char *ssleay_version_text = SSLeay_version(SSLEAY_VERSION);
+            EXPECT_NOT_NULL(strcasestr(ssleay_version_text, s2n_libcrypto_info.name));
+        }
+    };
+
+    /* Check libcrypto version matches the intent of the CI.  */
+    {
+        char intent_version[20] = { 0 };
+        if (convert_version_number_to_string(&s2n_libcrypto_info, intent_version) != -1) {
```

Comment:
Earlier when you parsed the version, you converted the string to a number. Now, when actually using the version, you convert it to a string. So I'm wondering, do you actually need to do this conversion at all?